### ISSUE TITLE
[v2] Crash in RtpStream::ResetHealthCheckTimer()

### ISSUE THREAD

**Issue by ibc** [versatica/mediasoup#146]
Not sure how to reproduce. I was testing a room with Chrome and Safari. Suddenly this happened:

```
Process:               mediasoup-worker [57659]
Path:                  /Users/USER/*/mediasoup-worker
Identifier:            mediasoup-worker
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        node [57658]
Responsible:           mediasoup-worker [57659]
User ID:               501

Date/Time:             2017-09-28 17:55:44.278 +0200
OS Version:            Mac OS X 10.12.6 (16G29)
Report Version:        12
Anonymous UUID:        2DAF0C89-78A7-6C53-2B77-1076907F53E1

Sleep/Wake UUID:       D0C7B3C9-89E7-48DC-9E8F-4528F1290145

Time Awake Since Boot: 270000 seconds
Time Since Wake:       4900 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00000000000000d9
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [0]

VM Regions Near 0xd9:
--> 
    __TEXT                 00000001067d8000-00000001069b6000 [ 1912K] r-x/rwx SM=COW  /Users/USER/*

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   mediasoup-worker              	0x0000000106806e62 RTC::RtpStream::ResetHealthCheckTimer(unsigned short) + 4 (RtpStream.cpp:122)
1   mediasoup-worker              	0x00000001067e6cc1 RTC::Consumer::RecalculateTargetProfile(bool) + 671 (Consumer.cpp:882)
2   mediasoup-worker              	0x00000001067e747e RTC::Consumer::SetPreferredProfile(RTC::RtpEncodingParameters::Profile) + 260 (Consumer.cpp:301)
3   mediasoup-worker              	0x00000001067fb82d RTC::Router::HandleRequest(Channel::Request*) + 7371 (Router.cpp:1213)
4   mediasoup-worker              	0x00000001067e4a4d Channel::UnixStreamSocket::UserOnUnixStreamRead() + 247 (UnixStreamSocket.cpp:295)
5   mediasoup-worker              	0x000000010686571c uv__stream_io + 999 (stream.c:1246)
6   mediasoup-worker              	0x000000010686c4d2 uv__io_poll + 1783 (kqueue.c:342)
7   mediasoup-worker              	0x000000010685e21c uv_run + 343 (core.c:363)
8   mediasoup-worker              	0x00000001067dfff4 Worker::Worker(Channel::UnixStreamSocket*) + 308 (Worker.cpp:37)
9   mediasoup-worker              	0x0000000106842832 main + 1474 (main.cpp:94)
10  libdyld.dylib                 	0x00007fffc78dd235 start + 1

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000002  rbx: 0x00007fe0291a3e30  rcx: 0x00007fe0293ab103  rdx: 0x00007fe0293ab160
  rdi: 0x0000000000000000  rsi: 0x00000000000003e8  rbp: 0x00007fff5941eb70  rsp: 0x00007fff5941eb70
   r8: 0x00000000293ab103   r9: 0x0000000000000004  r10: 0x0000000000000004  r11: 0x0000001f2f10780d
  r12: 0x000000000000009e  r13: 0x00007fe02a317460  r14: 0x00007fe0291a4019  r15: 0x00007fff5941ef20
  rip: 0x0000000106806e62  rfl: 0x0000000000010202  cr2: 0x00000000000000d9
  
Logical CPU:     2
Error Code:      0x00000006
Trap Number:     14


Binary Images:
       0x1067d8000 -        0x1069b5f87 +mediasoup-worker (0) <D0ED8895-A484-3752-B392-281614C53968> /Users/USER/*/mediasoup-worker
       0x1112c4000 -        0x111301dc7  dyld (433.5) <322C06B7-8878-311D-888C-C8FD2CA96FF3> /usr/lib/dyld
    0x7fffc6333000 -     0x7fffc6334ffb  libSystem.B.dylib (1238.60.2) <F18AC1E7-C6F1-34B1-8069-BE571B3231D4> /usr/lib/libSystem.B.dylib
    0x7fffc646d000 -     0x7fffc64c3ff7  libc++.1.dylib (307.5) <0B43BB5D-E6EB-3464-8DE9-B41AC8ED9D1C> /usr/lib/libc++.1.dylib
    0x7fffc64c4000 -     0x7fffc64edff7  libc++abi.dylib (307.4) <BC271AD3-831B-362A-9DA7-E8C51F285FE4> /usr/lib/libc++abi.dylib
    0x7fffc6fe2000 -     0x7fffc73b4047  libobjc.A.dylib (709.1) <70614861-0340-32E2-85ED-FE65759CDFFA> /usr/lib/libobjc.A.dylib
    0x7fffc7801000 -     0x7fffc7805ff7  libcache.dylib (79) <093A4DAB-8385-3D47-A350-E20CB7CCF7BF> /usr/lib/system/libcache.dylib
    0x7fffc7806000 -     0x7fffc7810fff  libcommonCrypto.dylib (60092.50.5) <8A64D1B0-C70E-385C-92F0-E669079FDA90> /usr/lib/system/libcommonCrypto.dylib
    0x7fffc7811000 -     0x7fffc7818fff  libcompiler_rt.dylib (62) <55D47421-772A-32AB-B529-1A46C2F43B4D> /usr/lib/system/libcompiler_rt.dylib
    0x7fffc7819000 -     0x7fffc7821fff  libcopyfile.dylib (138) <819BEA3C-DF11-3E3D-A1A1-5A51C5BF1961> /usr/lib/system/libcopyfile.dylib
    0x7fffc7822000 -     0x7fffc78a5fdf  libcorecrypto.dylib (442.50.19) <65D7165E-2E71-335D-A2D6-33F78E2DF0C1> /usr/lib/system/libcorecrypto.dylib
    0x7fffc78a6000 -     0x7fffc78d7fff  libdispatch.dylib (703.50.37) <6582BAD6-ED27-3B30-B620-90B1C5A4AE3C> /usr/lib/system/libdispatch.dylib
    0x7fffc78d8000 -     0x7fffc78ddffb  libdyld.dylib (433.5) <9B2AC56D-107C-3541-A127-9094A751F2C9> /usr/lib/system/libdyld.dylib
    0x7fffc78de000 -     0x7fffc78deffb  libkeymgr.dylib (28) <7AA011A9-DC21-3488-BF73-3B5B14D1FDD6> /usr/lib/system/libkeymgr.dylib
    0x7fffc78ec000 -     0x7fffc78ecfff  liblaunch.dylib (972.70.1) <B856ABD2-896E-3DE0-B2C8-146A6AF8E2A7> /usr/lib/system/liblaunch.dylib
    0x7fffc78ed000 -     0x7fffc78f2ff3  libmacho.dylib (898) <17D5D855-F6C3-3B04-B680-E9BF02EF8AED> /usr/lib/system/libmacho.dylib
    0x7fffc78f3000 -     0x7fffc78f5ff3  libquarantine.dylib (85.50.1) <12448CC2-378E-35F3-BE33-9DC395A5B970> /usr/lib/system/libquarantine.dylib
    0x7fffc78f6000 -     0x7fffc78f7ffb  libremovefile.dylib (45) <38D4CB9C-10CD-30D3-8B7B-A515EC75FE85> /usr/lib/system/libremovefile.dylib
    0x7fffc78f8000 -     0x7fffc7910ff7  libsystem_asl.dylib (349.50.5) <096E4228-3B7C-30A6-8B13-EC909A64499A> /usr/lib/system/libsystem_asl.dylib
    0x7fffc7911000 -     0x7fffc7911ff7  libsystem_blocks.dylib (67) <10DC5404-73AB-35B3-A277-A8AFECB476EB> /usr/lib/system/libsystem_blocks.dylib
    0x7fffc7912000 -     0x7fffc799ffef  libsystem_c.dylib (1158.50.2) <E5AE5244-7D0C-36AC-8BB6-C7AE7EA52A4B> /usr/lib/system/libsystem_c.dylib
    0x7fffc79a0000 -     0x7fffc79a3ffb  libsystem_configuration.dylib (888.60.2) <BECC01A2-CA8D-31E6-BCDF-D452965FA976> /usr/lib/system/libsystem_configuration.dylib
    0x7fffc79a4000 -     0x7fffc79a7fff  libsystem_coreservices.dylib (41.4) <7D26DE79-B424-3450-85E1-F7FAB32714AB> /usr/lib/system/libsystem_coreservices.dylib
    0x7fffc79a8000 -     0x7fffc79c0fff  libsystem_coretls.dylib (121.50.4) <EC6FCF07-DCFB-3A03-9CC9-6DD3709974C6> /usr/lib/system/libsystem_coretls.dylib
    0x7fffc79c1000 -     0x7fffc79c7fff  libsystem_dnssd.dylib (765.50.9) <CC960215-0B1B-3822-A13A-3DDE96FA796F> /usr/lib/system/libsystem_dnssd.dylib
    0x7fffc79c8000 -     0x7fffc79f1ff7  libsystem_info.dylib (503.50.4) <611DB84C-BF70-3F92-8702-B9F28A900920> /usr/lib/system/libsystem_info.dylib
    0x7fffc79f2000 -     0x7fffc7a14ff7  libsystem_kernel.dylib (3789.70.16) <34B1F16C-BC9C-3C5F-9045-0CAE91CB5914> /usr/lib/system/libsystem_kernel.dylib
    0x7fffc7a15000 -     0x7fffc7a5cfe7  libsystem_m.dylib (3121.6) <86D499B5-BBDC-3D3B-8A4E-97AE8E6672A4> /usr/lib/system/libsystem_m.dylib
    0x7fffc7a5d000 -     0x7fffc7a7bff7  libsystem_malloc.dylib (116.50.8) <A3D15F17-99A6-3367-8C7E-4280E8619C95> /usr/lib/system/libsystem_malloc.dylib
    0x7fffc7a7c000 -     0x7fffc7ad5ffb  libsystem_network.dylib (856.60.1) <369D0221-56CA-3C3E-9EDE-94B41CAE77B7> /usr/lib/system/libsystem_network.dylib
    0x7fffc7ad6000 -     0x7fffc7adfff3  libsystem_networkextension.dylib (563.60.2) <B021F2B3-8A75-3633-ABB0-FC012B8E9B0C> /usr/lib/system/libsystem_networkextension.dylib
    0x7fffc7ae0000 -     0x7fffc7ae9ff3  libsystem_notify.dylib (165.20.1) <B8160190-A069-3B3A-BDF6-2AA408221FAE> /usr/lib/system/libsystem_notify.dylib
    0x7fffc7aea000 -     0x7fffc7af2fe7  libsystem_platform.dylib (126.50.8) <897462FD-B318-321B-A554-E61982630F7E> /usr/lib/system/libsystem_platform.dylib
    0x7fffc7af3000 -     0x7fffc7afdff7  libsystem_pthread.dylib (218.60.3) <B8FB5E20-3295-39E2-B5EB-B464D1D4B104> /usr/lib/system/libsystem_pthread.dylib
    0x7fffc7afe000 -     0x7fffc7b01ff7  libsystem_sandbox.dylib (592.70.1) <4B92EC49-ACD0-36AE-B07A-A2B8152EAF9D> /usr/lib/system/libsystem_sandbox.dylib
    0x7fffc7b02000 -     0x7fffc7b03ff3  libsystem_secinit.dylib (24.50.4) <F78B847B-3565-3E4B-98A6-F7AD40392E2D> /usr/lib/system/libsystem_secinit.dylib
    0x7fffc7b04000 -     0x7fffc7b0bffb  libsystem_symptoms.dylib (532.50.47) <3390E07C-C1CE-348F-ADBD-2C5440B45EAA> /usr/lib/system/libsystem_symptoms.dylib
    0x7fffc7b0c000 -     0x7fffc7b1fff7  libsystem_trace.dylib (518.70.1) <AC63A7FE-50D9-3A30-96E6-F6B7FF16E465> /usr/lib/system/libsystem_trace.dylib
    0x7fffc7b20000 -     0x7fffc7b25ffb  libunwind.dylib (35.3) <3D50D8A8-C460-334D-A519-2DA841102C6B> /usr/lib/system/libunwind.dylib
    0x7fffc7b26000 -     0x7fffc7b4fff7  libxpc.dylib (972.70.1) <BF896DF0-D8E9-31A8-A4B3-01120BFEEE52> /usr/lib/system/libxpc.dylib

External Modification Summary:
  Calls made by other processes targeting this process:
    task_for_pid: 1
    thread_create: 0
    thread_set_state: 0
  Calls made by this process:
    task_for_pid: 0
    thread_create: 0
    thread_set_state: 0
  Calls made by all processes on this machine:
    task_for_pid: 207870
    thread_create: 0
    thread_set_state: 0

VM Region Summary:
ReadOnly portion of Libraries: Total=125.5M resident=0K(0%) swapped_out_or_unallocated=125.5M(100%)
Writable regions: Total=75.3M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=75.3M(100%)
 
                                VIRTUAL   REGION 
REGION TYPE                        SIZE    COUNT (non-coalesced) 
===========                     =======  ======= 
Kernel Alloc Once                    8K        2 
MALLOC                            66.3M       12 
MALLOC guard page                   16K        4 
STACK GUARD                       56.0M        2 
Stack                             8192K        2 
__DATA                            2428K       43 
__LINKEDIT                       115.8M        4 
__TEXT                            9936K       41 
shared memory                       12K        4 
===========                     =======  ======= 
TOTAL                            258.3M      105 

```

**Comment by ibc** [comment]
So can we close it?

**Comment by jmillan** [comment]
I thought it was automatically closed.

**Comment by jmillan** [comment]
ah.., probably the comment needed a `Fixed` instead of a `Fix`

**Comment by ibc** [comment]
It would be automatically closed if in master, or would be automatically closed once v2 is merged into master.

**Comment by jmillan** [comment]
Oh.., I didn't know that. Nice!
